src/prover_disk.hpp

```diff
@@ -241,6 +242,32 @@ class DiskProver {
     std::vector<uint64_t> table_begin_pointers;
     std::vector<uint64_t> C2;
 
+    // Using this method instead of simply seeking will prevent segfaults that would arise when
+    // continuing the process of looking up qualities.
+    void safeSeek(std::ifstream& disk_file, uint64_t seek_location) {
+        if ((disk_file.rdstate() & std::ifstream::goodbit) != 0) {
+            throw std::invalid_argument("safeSeed: provided input stream goodbit != 0.");
+        }
+
+        disk_file.seekg(seek_location);
```

Comment:
Seeking beyond end-of-file is not an error. I think it's necessary to only wrap calls to `read`, but not to `seekg`.

Reply:
Are you saying seeking beyond the end will not trigger a badbit or failbit? It does under my tests.
Basically an invalid pointer read from disk (bad plot) will cause seek to set badbit or failbit.

Reply:
We should check the error bits set after `.read()` in `safeRead`. Checking them once again here is not necessary. I'm still reasonably sure that `.seekg()` will not set error bits by itself, since seeking beyond EOF is a valid operation. Which OS did you test it on?